docs/server-side-rendering.md

```diff
@@ -0,0 +1,154 @@
+
+```
+npm install --save-dev express webpack-cli ts-loader ws xmlhttprequest
+```
+
+Create a file called `server.ts` in the root of you project.
+
+```ts
+// These are important and needed before anything else
+import 'zone.js/dist/zone-node';
+import 'reflect-metadata';
+
+import { renderModuleFactory } from '@angular/platform-server';
+import { enableProdMode } from '@angular/core';
+
+import * as express from 'express';
+import { join } from 'path';
+import { readFileSync } from 'fs';
+
+// Required for Firebase
+(global as any).WebSocket = require('ws');
+(global as any).XMLHttpRequest = require('xmlhttprequest').XMLHttpRequest;
+
+
+// Faster renders in prod mode
+enableProdMode();
+
+// Express server
+const app = express();
+
+const PORT = process.env.PORT || 4000;
+const DIST_FOLDER = join(process.cwd(), 'dist');
+const APP_NAME = 'YOUR_PROJECT_NAME';
+
+const { AppServerModuleNgFactory } = require(`./dist/${APP_NAME}-server/main`);
+
+// index.html template
+const template = readFileSync(join(DIST_FOLDER, APP_NAME, 'index.html')).toString();
+
+app.engine('html', (_, options, callback) => {
+  renderModuleFactory(AppServerModuleNgFactory, {
+    document: template,
+    url: options.req.url,
+  }).then(html => {
+    callback(null, html);
+  });
+});
+
+app.set('view engine', 'html');
+app.set('views', join(DIST_FOLDER, APP_NAME));
+
+// Serve static files 
+app.get('*.*', express.static(join(DIST_FOLDER, APP_NAME)));
+
+// All regular routes use the Universal engine
+app.get('*', (req, res) => {
+    res.render(join(DIST_FOLDER, APP_NAME, 'index.html'), { req });
+});
+
+// Start up the Node server
+app.listen(PORT, () => {
+  console.log(`Node server listening on http://localhost:${PORT}`);
+});
+```
+
+## 3. Add a Webpack Config for the Express Server
+
+Create a new file named `webpack.server.config.js` to bundle the express app from previous step. 
+
+
+```js
+const path = require('path');
+const webpack = require('webpack');
+
+const APP_NAME = 'YOUR_PROJECT_NAME';
+
+module.exports = {
+  entry: {  server: './server.ts' },
+  resolve: { extensions: ['.js', '.ts'] },
+  mode: 'development',
+  target: 'node',
+  externals: [/(node_modules|main\..*\.js)/],
+  output: {
+    path: path.join(__dirname, `dist/${APP_NAME}`),
+    filename: '[name].js'
+  },
+  module: {
+    rules: [
+      { test: /\.ts$/, loader: 'ts-loader' }
+    ]
+  },
+  plugins: [
+    new webpack.ContextReplacementPlugin(
+      /(.+)?angular(\\|\/)core(.+)?/,
+      path.join(__dirname, 'src'), // location of your src
+      {} // a map of your routes
+    ),
+    new webpack.ContextReplacementPlugin(
+      /(.+)?express(\\|\/)(.+)?/,
+      path.join(__dirname, 'src'),
+      {}
+    )
+  ]
+}
+```
+
+## 4.0 Build Scripts
+
+Update your `package.json` with the following build scripts. 
+
+```js
+"scripts": {
+  // ... omitted
+  "build:ssr": "ng build --prod && ng run YOUR_PROJECT_NAME:server && npm run webpack:ssr",
```

Comment:
Isn't it worth mentioning that `ng run YOUR_PROJECT_NAME:server` will actually use the dev configuration, i.e. `environment.ts` and not `environment.prod.ts`, while `ng build --prod` is actually building the client for production?